webapp/test/NewsList.test.js

```diff
@@ -46,51 +101,42 @@ require('regenerator-runtime/runtime');
     const button = wrapper.find('button.sorter');
     button.trigger('click');
     expect(wrapper.vm.descending).toEqual(false);
-    expect(wrapper.vm.sortedItems[0]).toEqual({ title: "Title_2", votes: 4})
+    expect(wrapper.vm.sortedItems[0]).toEqual({ title: "Title_2", votes: 4, author: { id: 1}})
   })
 
   it("should add an item to its internal array when the user clicks the submit button and provides a valid title",  async () => {
-      let newsForm = wrapper.findComponent(NewsForm);
-      newsForm.setData({
-          valid: false,
-          newsTitle: 'Title_3',
-      });
-      
-      newsForm.find('v-btn').trigger('click');
-
-      expect(newsForm.emitted('addItem')).toBeTruthy();
-      expect(newsForm.emitted('addItem').length).toBe(1);
-
-      expect(wrapper.vm.newsItems.pop()).toEqual({title: "Title_3", votes: 0});
+    const testWrapper = setupWrapper({
+      store,
+      localVue,
+      mocks: {
+        $apollo: {
+          query: queryMock,
+          mutate: writeMutateMock
+        }
+      }
+    }, shallowMount)
+    await testWrapper.vm.addItem('Title_3');
+    expect(writeMutateMock).toBeCalled()
+    expect(queryMock).toHaveBeenCalledTimes(2); // After writing, we also expect vue to rerender the component and therefore query the news again
```

Comment:
Ideally, this is not necessary. Once the `write` mutation completes, it adds the new `post` to the cached `posts` query, thus triggering a re-render *without unnecessary graphql queries*.